packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -80,35 +84,35 @@ exports.onPostBuild = (args, pluginOptions) => {
     // URLs and not any files hosted on the site.
     //
     // Regex based on http://stackoverflow.com/a/18017805
-    navigateFallbackWhitelist: [/^.*([^.]{5}|.html)(?<!(\?|&)no-cache=1)$/],
+    navigateFallbackWhitelist: [/^[^?]*([^.?]{5}|\.html)(\?.*)?$/],
+    navigateFallbackBlacklist: [/\?(.+&)?no-cache=1$/],
     cacheId: `gatsby-plugin-offline`,
     // Don't cache-bust JS files and anything in the static directory
     dontCacheBustUrlsMatching: /(.*js$|\/static\/)/,
     runtimeCaching: [
       {
         // Add runtime caching of various page resources.
         urlPattern: /\.(?:png|jpg|jpeg|webp|svg|gif|tiff|js|woff|woff2|json|css)$/,
-        handler: `fastest`,
+        handler: `staleWhileRevalidate`,
```

Comment:
In the future, we can be smarter here e.g. our js/css files + anything in the static folder never needs to be revalidated.

Reply:
Maybe we could just add `.*\.css` to the `dontCacheBustUrlsMatching` regex? (Also I just noticed there should be a `\.` before `js` in that regex to prevent edge cases)

Reply:
Yeah that'd be good.

But we could also do a cache only option for those. Though perhaps doesn't matter as the browser cache should take care of the requests I believe.